pkg/controllers/disruption/emptynodeconsolidation.go

```diff
@@ -37,8 +39,9 @@ type EmptyNodeConsolidation struct {
 }
 
 func NewEmptyNodeConsolidation(clk clock.Clock, cluster *state.Cluster, kubeClient client.Client,
-	provisioner *provisioning.Provisioner, cp cloudprovider.CloudProvider, recorder events.Recorder) *EmptyNodeConsolidation {
-	return &EmptyNodeConsolidation{consolidation: makeConsolidation(clk, cluster, kubeClient, provisioner, cp, recorder)}
+	provisioner *provisioning.Provisioner, cp cloudprovider.CloudProvider, recorder events.Recorder,
+	queue *orchestration.Queue) *EmptyNodeConsolidation {
+	return &EmptyNodeConsolidation{consolidation: makeConsolidation(clk, cluster, kubeClient, provisioner, cp, recorder, queue)}
```

Comment:
Why not call `makeConsolidation` outside of this function? Seems like an awful lot of wiring / leaky abstraction

Reply:
Good callout, we could just re-factor this to wrap each consolidation type with a constructed consolidation in the disruption controller constructor. I can include this in the stage of changes to merge before this.